keps/sig-node/2008-forensic-container-checkpointing/README.md

```diff
@@ -244,25 +315,62 @@ We expect no non-infra related flakes in the last month as a GA graduation crite
   Once CRI implementation provide the relevant RPC calls
   the e2e tests will not fail but need to be extended.
 
+- Once the initial Alpha release  CRI-O supports the
+  `CheckpointContainer` CRI RPC and tests have been
+  enhanced to support CRI implementation that implement
+  the `CheckpointContainer` CRI RPC
+
+- Once Kubernetes was released with the `CheckpointContainer` CRI RPC
+  CRI-O has been updated to support the new CRI RPC.
+  The tests have been enhanced to work with CRI implementations
+  that support the `CheckpointContainer` CRI RPC as well as
+  CRI implementations that do not support it. The tests also handle
+  if the corresponding feature gate is disabled or enabled:
+  <https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/checkpoint_container.go>
+
+- As the tests are hidden behind the feature gate `ContainerCheckpoint` during
+  Alpha phase and only available in combination with CRI-O automatic, the tests
+  have been skipped so far. With graduation to Beta the tests should appear in
+  CRI-O based setups. Due to way the current setup of not running all Alpha
+  features enabled with CRI-O, no results have been collected and tests have
+  been skipped.
+
 ### Graduation Criteria
 
 #### Alpha
 
-- [ ] Implement the new feature gate and kubelet implementation
-- [ ] Ensure proper tests are in place
-- [ ] Update documentation to make the feature visible
+- [X] Implement the new feature gate and kubelet implementation
+- [X] Ensure proper tests are in place
+- [X] Update documentation to make the feature visible
+  - <https://kubernetes.io/docs/reference/node/kubelet-checkpoint-api/>
+  - <https://kubernetes.io/blog/2022/12/05/forensic-container-checkpointing-alpha/>
+  - <https://kubernetes.io/blog/2023/03/10/forensic-container-analysis/>
 
 #### Alpha to Beta Graduation
 
-At least one container engine has to have implemented the
-corresponding CRI APIs to introduce e2e test for checkpointing.
+At least one container engine implemented the corresponding CRI APIs:
+
+- [x] CRI-O
+
+In Kubernetes:
 
+- [x] No major bugs reported in the previous cycle
 - [ ] Enable the feature per default
-- [ ] No major bugs reported in the previous cycle
+- [ ] Add necessary metrics as described in the PRR sections
+  - [ ] Add CRI metrics
```

Comment:
I talked a bit with @haircommander about these metrics.

I think you could add a kubelet metric that detects success/failures/duration of the container checkpoint CRI call.

The two metrics that could be used as an example:

```
	// Metrics keys for RuntimeClass
	RunPodSandboxDurationKey = "run_podsandbox_duration_seconds"
	RunPodSandboxErrorsKey   = "run_podsandbox_errors_total"
```

I'm not sure if we have metrics around the kubelet api at the moment.